cachet-tools/cachet_update_daemon.py

```diff
@@ -0,0 +1,93 @@
+"""Cachet Update Daemon
+This script is a daemon that handles incoming webhooks from Alertmanager.
+Each webhook should represent a test suite for a component represented
+in the Cachet status page.
+
+The only configuration that needs to be done is to export the cachet
+URL and the cachet X token as ENV variables.  Example:
+
+export CACHET_URL="cachet.cachet.svc"
+export CACHET_TOKEN="fXPydfUD3c1aBUGiQnSb"
+"""
+import json
+import os
+import logging
+import requests
+from flask import Flask, request
+
+APP = Flask(__name__)
+CACHET_IDS = ""
+CACHET_URL = os.getenv("CACHET_URL")
+CACHET_TOKEN = os.getenv("CACHET_TOKEN")
+logging.basicConfig(level='INFO', format='%(asctime)s %(message)s', datefmt='%m/%d/%Y %I:%M:%S %p')
+logger = logging.getLogger(__name__)
+
+@APP.route('/', methods=['POST'])
+def handle_alert():
+    """ This function handles incoming post calls from the Alertmanager
+    webhook. After receiving the post from AM it reaches out to Cachet
+    and builds a map of it's components. Once the components are
+    searched and the component from the alert is located it updates said
+    component with the proper status.
+    """
+    alert_in = json.loads(request.data)
+    logging.info("Dumping incoming JSON from Alertmanager:")
+    logging.info(str(json.dumps(alert_in, indent=4)))
```

Comment:
this should probably be a debug log statement

Reply:
also - why do you have to cast the string that comes out of `json.dumps` to a string?

Reply:
Force of habit, I'll change.